### PR TITLE
Red-369 React component

### PR DESCRIPTION
**Pull request by au-e2xlabs**

### CODE REVIEW

File: src/components/DropDown.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      {loading ? <StyledLoader type="spinningBubbles" color={baseStyle.primaryColor} height={baseStyle.loader.height}
+                               width={baseStyle.loader.height}/> : null}
+      {!loading && content.length > 0 ? <UnorderedList>
+        {content.sort((a, b) => a.network.localeCompare(b.network)).map((item, key) => {
+          return (
+            <ListItem key={key}>
+              <ItemWrapper>
+                <StyledContent>
+                  <StyledLogo width={baseStyle.dropDown.logo.width} src={CoinLogos[item.network.toLocaleUpperCase()]}
+                              alt="coinLogo"/>
```
Comment (Ivan-dev-RED) [inline src/components/DropDown.tsx:37]: Logo not clickable?
Let`s do the full field clickable but prevent the mdiContentCopy click

File: src/components/RedefinedDomainResolver.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+const RedefinedDomainResolver = (props: RedefinedDomainResolverProps) => {
+  const {width, height, onSelect} = props;
+  const [dropDownActive, setDropDownActive] = useState(false);
+  const [domain, setDomain] = useState("");
+  const [addresses, setAddresses] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState("");
+
+  useEffect(() => {
+    domain.length === 0 && setDropDownActive(false);
```
Comment (Ivan-dev-RED) [inline src/components/RedefinedDomainResolver.tsx:20]: It is good if we will return sth `return  domain.length === 0 && getShit()`
But without return it is not good code style for ts/js, i think

File: src/components/RedefinedDomainResolver.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  }, [domain])
+
+  const handleDebounceFn = async (value: string) => {
+    if (value.length > 0) {
+      setDropDownActive(true);
+      setLoading(true);
+      setAddresses([]);
+      setError("");
+      await new RedefinedResolver().resolve(value)
```
Comment (Ivan-dev-RED) [inline src/components/RedefinedDomainResolver.tsx:29]: try catch?😅

File: src/components/RedefinedDomainResolver.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  const [domain, setDomain] = useState("");
+  const [addresses, setAddresses] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState("");
+
+  useEffect(() => {
+    domain.length === 0 && setDropDownActive(false);
+  }, [domain])
+
+  const handleDebounceFn = async (value: string) => {
```
Comment (Ivan-dev-RED) [inline src/components/RedefinedDomainResolver.tsx:23]: resolveDomain

File: src/components/RedefinedDomainResolver.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+      setAddresses([]);
+      setError("");
+      await new RedefinedResolver().resolve(value)
+        .then((res) => setAddresses(res))
+        .catch(error => setError(error));
+      setLoading(false);
+    }
+  }
+
+  const fetchAddresses = useCallback(_debounce(handleDebounceFn, 500), []);
```
Comment (Ivan-dev-RED) [inline src/components/RedefinedDomainResolver.tsx:36]: resolveDomainWithDebounce

File: src/utils.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+export const copyText = (text: string): void => {
+  navigator.clipboard.writeText(text).catch(function (err) {
+    console.error('Could not copy text: ', err);
+  });
+};
+
+export const getAbbreviatedAddress = (address: string): string => (
+  `${address.substring(0, 12)}...${address.substring(address.length - 5, address.length)}`
```
Comment (Ivan-dev-RED) [inline src/utils.ts:14]: 12 - too much i think

`${address.substring(0, 6)} ... ${address.substring(address.length - 4, address.length)}`

**Comment by Ivan-dev-RED** [comment]
To fix build you need to install `@redefined/name-resolver-js`

File: src/components/DropDown.tsx (hunk 1 above)
Comment (au-e2xlabs) [reply src/components/DropDown.tsx:37]: I don't see the point in that

File: src/components/RedefinedDomainResolver.tsx (hunk 2 above)
Comment (au-e2xlabs) [reply src/components/RedefinedDomainResolver.tsx:20]: what is `getShift()?`  why is return here? i don`t understand.

File: src/utils.ts (hunk 6 above)
Comment (au-e2xlabs) [reply src/utils.ts:14]: i try it. but i see that result if input width = 350px. I think we can pass more information. 
image

File: src/components/DropDown.tsx [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+import ReactLoading from "react-loading";
+import {baseStyle} from "../styles/baseStyle";
+
+const DropDown = (props: DropdownProps) => {
+  const {active, content, loading, error, onChange, onClickOutside} = props;
+  const ref = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (ref.current && !ref.current.contains(event.target)) {
```
[the remark this answers is not in the document]
Comment (au-e2xlabs) [reply src/components/DropDown.tsx:16]: done

File: src/components/RedefinedDomainResolver.tsx (hunk 4 above)
Comment (au-e2xlabs) [reply src/components/RedefinedDomainResolver.tsx:23]: done

File: src/components/DropDown.tsx (hunk 1 above)
Comment (Ivan-dev-RED) [reply src/components/DropDown.tsx:37]: Okay, we will look on it tomorrow
May be it`s a problem only for me)

File: src/components/RedefinedDomainResolver.tsx (hunk 2 above)
Comment (Ivan-dev-RED) [reply src/components/RedefinedDomainResolver.tsx:20]: getShift is an example)
It does not return here, but this entry `return  domain.length === 0 && getShit()` implies a return
Let`s use this if you want to do it in line)
```
if (domain.length === 0) setDropDownActive(false);
```

File: src/components/RedefinedDomainResolver.tsx (hunk 3 above)
Comment (Ivan-dev-RED) [reply src/components/RedefinedDomainResolver.tsx:29]: Fii
`then catch` is not actual and in this case you can write without them)

File: src/utils.ts (hunk 6 above)
Comment (Ivan-dev-RED) [reply src/utils.ts]: Fine
Let's use your choice
And in the "custom styles support" ticket, we need to support custom address formatting

File: src/components/RedefinedDomainResolver.tsx (hunk 3 above)
Comment (au-e2xlabs) [reply src/components/RedefinedDomainResolver.tsx:29]: ok done

**Review by Ivan-dev-RED — APPROVED** [review]

**Pull request merged**